aiozk/connection.py

```diff
@@ -174,15 +174,18 @@ def handle_write(write_future):
 
     async def _read(self, size=-1):
         remaining_size = size
-        payload = b''
         end_time = time() + self.read_timeout
+        payload = []
         while remaining_size and (time() < end_time):
-            chunk = await self.reader.read(remaining_size)
-            payload += chunk
-            remaining_size -= len(chunk)
+            remaining_time = end_time - time()
+            done, _ = await asyncio.wait([self.reader.read(remaining_size)], timeout=remaining_time)
```

Comment:
Oh, I've realized that here we have to close not finished futures here (assigned `_` in your code).
Or we can use `asyncio.wait_for` + handle `asyncio.TimeoutError`